addons/web/static/src/webclient/menus/menu_service.js

```diff
@@ -21,69 +18,85 @@ function makeFetchLoadMenus() {
     };
 }
 
-function makeMenus(env, menusData, fetchLoadMenus) {
-    let currentAppId;
-    function _getMenu(menuId) {
-        return menusData[menuId];
-    }
-    function setCurrentMenu(menu) {
-        menu = typeof menu === "number" ? _getMenu(menu) : menu;
-        if (menu && menu.appID !== currentAppId) {
-            currentAppId = menu.appID;
-            browser.sessionStorage.setItem("menu_id", currentAppId);
-            env.bus.trigger("MENUS:APP-CHANGED");
-        }
-    }
+export const menuService = {
+    dependencies: ["action"],
+    async start(env) {
+        let currentAppId;
+        let menusData;
 
-    return {
-        getAll() {
-            return Object.values(menusData);
-        },
-        getApps() {
-            return this.getMenu("root").children.map((mid) => this.getMenu(mid));
-        },
-        getMenu: _getMenu,
-        getCurrentApp() {
-            if (!currentAppId) {
-                return;
-            }
-            return this.getMenu(currentAppId);
-        },
-        getMenuAsTree(menuID) {
-            const menu = this.getMenu(menuID);
-            if (!menu.childrenTree) {
-                menu.childrenTree = menu.children.map((mid) => this.getMenuAsTree(mid));
-            }
-            return menu;
-        },
-        async selectMenu(menu) {
-            menu = typeof menu === "number" ? this.getMenu(menu) : menu;
-            if (!menu.actionID) {
-                return;
-            }
-            await env.services.action.doAction(menu.actionID, {
-                clearBreadcrumbs: true,
-                onActionReady: () => {
-                    setCurrentMenu(menu);
-                },
+        const fetchLoadMenus = makeFetchLoadMenus();
```

Comment:
why do we need this `make` mechanism? Can't we simply have a `fetchMenus` function above?

Reply:
We could not before, but now I think we can